include/safe/detail/concepts.hpp

```diff
@@ -15,4 +15,35 @@ concept range_like = requires(R r) {
     { r.begin() } -> iter_like;
     r.end();
 };
-} // namespace safe::detail
\ No newline at end of file
+
+template <char Char>
+constexpr bool is_decimal_digit_v = Char - '0' >= 0 && Char - '0' <= 9;
+
+template <typename T, char... Chars>
+concept positive_decimal_integer = requires { requires std::integral<T>; } &&
+                                   (is_decimal_digit_v<Chars> && ...);
```

Comment:
**issue**: Rather than using a nested concept, just put the concept itself. So:
```cpp
template <typename T, char... Chars>
concept positive_decimal_integer = std::integral<T> &&
                                   (is_decimal_digit_v<Chars> && ...);
```
That will enable the compiler to see subsumption, which means easier overload sets.